src/storage/mod.rs

```diff
@@ -149,6 +157,9 @@ pub trait Storage<I>:
     fn get<T>(&self, id: &I) -> &T::Archived
     where
         T: Archive;
+
```

Comment:
I don't think it makes sense to have the persist as part of the trait, it should be directly on the HostStore. Not all stores will be supporting persistance.